rust/numaflow-core/src/pipeline/isb/jetstream/reader.rs

```diff
@@ -92,11 +112,23 @@ impl TryFrom<JSWrappedMessage> for Message {
             value.partition_idx,
         ));
 
+        let body = match value.compression_type {
+            Some(CompressionType::Gzip) => {
+                let mut decoder: GzDecoder<&[u8]> = GzDecoder::new(body.payload.as_ref());
```

Comment:
When I run the pipeline with compression, I see that sometimes the `body.payload` is empty (happens 1 or 2 times every minute). This is causing the decoder to fail.